### PR TITLE
refactoring refreshing of access token

### PR DESCRIPTION
**Pull request by TONYOG12**

### CODE REVIEW

File: app/core/security.py [hunk 1]
```diff
@@ -36,6 +36,23 @@ def create_token(self, user_id):
         )
 
         return token
+    
+
+    def create_refresh_token(self, user_id):
```
[the remark this answers is not in the document]
Comment (TONYOG12) [reply app/core/security.py:41]: alright, then I should prolly use the create_access_token() function from flask jwt extended as well

File: app/core/settings.py [hunk 2]
```diff
@@ -16,6 +16,7 @@ class Settings(BaseSettings):
     API_STR: str = "/jfarms"
     SECRET_KEY: str
     JWT_SECRET_KEY: str
+    REFRESH_TOKEN_COOKIE_KEY_NAME: str
```
[the remark this answers is not in the document]
Comment (TONYOG12) [reply app/core/settings.py:19]: noted

File: app/ents/employee/endpoints.py [hunk 3]
```diff
@@ -77,25 +83,49 @@ def login_employee():
...
     try:
+        refresh_token = request.cookies.get(settings.REFRESH_TOKEN_COOKIE_KEY_NAME)
         employee_id = str(get_jwt_identity())
-        if employee_id:
-            expiresIn = get_jwt()["exp"]
-            new_access_token = security.refresh_token(expiresIn, employee_id)
-            set_access_cookies(success_response, new_access_token)
-        return success_response
 
+        if refresh_token and employee_id:
```
[the remark this answers is not in the document]
Comment (TONYOG12) [reply app/ents/employee/endpoints.py]: latest commit should you resolve this

File: app/ents/employee/endpoints.py [hunk 4]
```diff
@@ -77,25 +84,40 @@ def login_employee():
 
     employee = authenticate(crud, email, password)
     if employee:
-        return success_response(
+        access_token = create_access_token(employee.id, expires_delta=timedelta(minutes=settings.ACCESS_TOKEN_EXPIRE_MINUTES))
+        refresh_token = create_refresh_token(employee.id, expires_delta=timedelta(minutes=settings.REFRESH_TOKEN_EXPIRE_MINUTES))
+
+        resp= success_response(
```
Comment (CorneliusOsei2) [inline app/ents/employee/endpoints.py:90]: How about setting the response cookie online 90 and then returning success_response after that? Just to simplify the code

Comment (CorneliusOsei2) [reply app/ents/employee/endpoints.py:90]: Review and merge my current PR.
success_response now returns a Flask Response object. You can use the set_cookie method on that returned object to set the refresh token in the cookie

Comment (TONYOG12) [reply app/ents/employee/endpoints.py:75]: okay great!

**Review by CorneliusOsei2 — APPROVED** [review]

**Comment by CorneliusOsei2** [comment]
Great job!

**Pull request merged**